debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/olr/OpenLogReplicatorStreamingChangeEventSource.java

```diff
@@ -0,0 +1,631 @@
+            return Optional.empty();
+        }
+
+        dispatcher.dispatchSchemaChangeEvent(
+                partition,
+                offsetContext,
+                tableId,
+                new OracleSchemaChangeEventEmitter(
+                        connectorConfig,
+                        partition,
+                        offsetContext,
+                        tableId,
+                        tableId.catalog(),
+                        tableId.schema(),
+                        tableDdl,
+                        schema,
+                        Instant.now(),
+                        streamingMetrics,
+                        null));
+
+        return Optional.ofNullable(schema.tableFor(tableId));
+    }
+
+    private void processTruncateEvent(StreamingEvent event, SchemaChangeEvent ddlEvent) throws InterruptedException {
+        if (ddlEvent.getSchema() == null) {
+            LOGGER.warn("Truncate event ignored, no schema found.");
+            return;
+        }
+
+        final TableId tableId = ddlEvent.getSchema().getTableId(event.getDatabaseName());
+        if (!connectorConfig.getTableFilters().dataCollectionFilter().isIncluded(tableId)) {
+            LOGGER.warn("Truncate event ignored, table is no included.");
+            return;
+        }
+
+        Table table = schema.tableFor(tableId);
+        if (table == null) {
+            try {
+                Optional<Table> result = potentiallyEmitSchemaChangeForUnknownTable(ddlEvent.getType(), tableId);
+                if (result.isEmpty()) {
+                    LOGGER.warn("Truncate ignored, cannot find table relational model");
+                    return;
+                }
+                table = result.get();
+            }
+            catch (Exception e) {
+                LOGGER.warn("Truncate ignored, failed to emit schema change", e);
+                return;
+            }
+        }
+
+        final Instant timestamp = Instant.ofEpochMilli(Long.parseLong(event.getTimestamp()));
+
+        offsetContext.setScn(Scn.valueOf(event.getScn()));
+        offsetContext.setEventScn(Scn.valueOf(event.getScn()));
+        offsetContext.setTransactionId(event.getXid());
+        offsetContext.tableEvent(tableId, timestamp);
+
+        LOGGER.trace("Dispatching {} (SCN {}) for table {}", Operation.TRUNCATE, event.getScn(), tableId);
+        dispatcher.dispatchDataChangeEvent(
+                partition,
+                tableId,
+                new OpenLogReplicatorChangeRecordEmitter(
+                        connectorConfig,
+                        partition,
+                        offsetContext,
+                        Operation.TRUNCATE,
+                        new Object[table.columns().size()],
+                        new Object[table.columns().size()],
+                        table,
+                        schema,
+                        clock));
+    }
+
+    private Object resolveColumnValue(TableId tableId, Column column, Values values) {
```

Comment:
Should not this be handled via `OracleValueConverter` or a subcalss of it?

Reply:
I'll see what can be done on this.  We'll definitely need to keep the `UNAVAILABLE_VALUE` loop in the event source handler as the value converters expect to look for this marker value and do the substitution; so I'll focus primarily on the _switch_ statement bit being wrapped into the value converter.